### DISCUSSION TITLE
How to make the UI transparent?

### DISCUSSION THREAD

**Discussion by sNishant011** [LazyVim/LazyVim#116] [Q&A] [answered]
I tried making UI transparent by adding following command in `autocmds.lua` file.
```lua
vim.api.nvim_create_autocmd("VimEnter", {
  callback = function()
    vim.cmd("hi! Normal guibg=NONE ctermbg=NONE")
    vim.cmd("hi! NonText guibg=NONE ctermbg=NONE")
  end,
})
```
I have following two problems using this:
1. There is no transparent background in `neo-tree`.
2. When the focus goes on to `neo-tree` the buffer transparency is lost.

**Comment by folke** [comment, marked as answer]
```lua
  {
    "tokyonight.nvim",
  opts = {
        transparent = true,
        styles = {
           sidebars = "transparent",
           floats = "transparent",
        },
  }
}
```

**Reply by kev-er** [reply]
Hi I am new to NeoVim,  how do I set this up?

**Reply by sNishant011** [reply]
@kev-er Setup Lazy vim first as per the documentation and then create `tokyonight.lua` file inside `plugins` directory of `nvim` config then add the given contents in that file.

**Reply by folke** [reply]
Create a file like `~/.config/nvim/lua/plugins/colorscheme.lua` and add `return ` followed by the code I posted above

**Reply by finxlfantasy** [reply]
@folke This isn't working on my macbook, i've tried adding transparency on both tokyonight and catppuccin, not sure what's the problem? I'm a noob to nvim.

**Reply by finxlfantasy** [reply]
Nevermind managed to fix it it. Fault was iterm2 not having the transparency settings on.

**Reply by pkazmier** [reply]
Could you clarify what you mean by transparent?

For example, setting a vim theme to be transparent means the background you see in vim will be that of your underlying terminal background color. If you want complete transparency to your desktop, you need to make sure you use a terminal that offers transparency as well. Here is a screenshot of my setup using WezTerm:

![Screenshot 2023-08-01 at 12 55 18 PM](https://github.com/LazyVim/LazyVim/assets/747855/11c1ecae-bef7-404f-ad55-97dad885c28e)

**Reply by Z1xonu4a** [reply]
cool

**Reply by mbougarne** [reply]
But it doesn't work with Twilight theme.

**Reply by TheArcadeCrow** [reply]
I am very new to nvim and am confused on where I have gone wring here. Is there anything else you need to do other then creating this file? for clarification the only thing I have done currently to make nvim transparent is create this file and paste in the code you provided.

**Reply by Subas-mohanty** [reply]
How to set a custom opacity for transparency ?

**Reply by SheepyChan** [reply]
> ```lua
>   {
>     "tokyonight.nvim",
>   opts = {
>         transparent = true,
>         styles = {
>            sidebars = "transparent",
>            floats = "transparent",
>         },
>   }
> }
> ```

does this only work on "tokyonight" theme?

**Comment by MuhammadHassanminhas** [comment]
Simply go in to your   **init.lua**   file and just simply paste these lines of code  there :

```
function Transparent(color)
color = color or "tokyonight"
 vim.cmd.colorscheme(color)
vim.api.nvim_set_hl(0, "Normal", { bg = "none" })
 vim.api.nvim_set_hl(0, "NormalFloat", { bg = "none" })
end
Transparent()
```

In these lines of code **function Transparent(color)** is a global function 
so it will execute every time when you open up your neovim.

You can change color theme by just simply replacing  ***tokyonight*** in the ` color = color or "tokyonight"` with your desired color scheme.
### THIS WILL CHANGE YOUR NEOVIM SETUP FROM THIS
![Screenshot from 2024-03-19 15-11-32](https://github.com/LazyVim/LazyVim/assets/140333751/163aeae5-75cb-4d65-addd-c6a2a8e117b4)
### TO THIS
![Screenshot from 2024-03-19 15-19-32](https://github.com/LazyVim/LazyVim/assets/140333751/426d6c36-de37-498f-b24a-54cbc1343573)

**Reply by EthanNguyenPM** [reply]
Can I get the background with spaceman xD?

**Reply by thecatthatmeows** [reply]
NICE! this is what i was looking for :)

**Reply by sansroot** [reply]
working perfectly, thanks!

**Reply by Alilacream** [reply]
Working thank you :D

**Reply by MuhammadHassanminhas** [reply]
> Can I get the background with spaceman xD?

<img width="1672" height="941" alt="background_img" src="https://github.com/user-attachments/assets/29f87fe3-6249-4d73-a94f-02cc8f6fb18c" />

**Comment by gmarxcc** [comment]
An updated to the correct answer that works fine in the newest nvim version:

The file to create is  `~/.config/nvim/lua/plugins/colorscheme.lua` and the contents of the file is:

```
return {
  "tokyonight.nvim",
  lazy = true,
  priority = 1000,
  opts = function()
    return {
      transparent = true,
    }
  end,
}
```

**Reply by Agus9217** [reply]
work with neo-tree background?

**Reply by Rouzihiro** [reply]
-- Bootstrap lazy.nvim and LazyVim
require("config.lazy")

-- Function to apply transparency settings globally
local function set_transparency()
  vim.cmd([[
    hi Normal guibg=NONE ctermbg=NONE
    hi NormalNC guibg=NONE ctermbg=NONE
    hi SignColumn guibg=NONE ctermbg=NONE
    hi StatusLine guibg=NONE ctermbg=NONE
    hi StatusLineNC guibg=NONE ctermbg=NONE
    hi VertSplit guibg=NONE ctermbg=NONE
    hi TabLine guibg=NONE ctermbg=NONE
    hi TabLineFill guibg=NONE ctermbg=NONE
    hi TabLineSel guibg=NONE ctermbg=NONE
    hi Pmenu guibg=NONE ctermbg=NONE
    hi PmenuSel guibg=NONE ctermbg=NONE
    hi NeoTreeNormal guibg=NONE ctermbg=NONE
    hi NeoTreeNormalNC guibg=NONE ctermbg=NONE
    hi NeoTreeWinSeparator guibg=NONE ctermbg=NONE
  ]])
end

-- Apply transparency settings initially
set_transparency()

-- Set up neo-tree with default settings and no custom mappings
require('neo-tree').setup({
  window = {
    -- Use default settings with no custom mappings
  },
})

-- Reapply transparency on buffer enter
vim.api.nvim_create_autocmd("BufEnter", {
  pattern = "*",
  callback = set_transparency,
})

**Reply by Rouzihiro** [reply]
this worked for me and created transparency also to Neo-tree

**Reply by josvaal** [reply]
to make it completely transparent would be to add this more::

```vim
hi NeoTreeEndOfBuffer guibg=NONE ctermbg=NONE
hi EndOfBuffer guibg=NONE ctermbg=NONE
```

**Reply by MDSADABWASIM** [reply]
This is the final command you need to add in `init.lua` to make everything transparent: 

```
-- bootstrap lazy.nvim, LazyVim and your plugins
require("config.lazy")

-- Function to apply transparency settings globally
local function set_transparency()
  vim.cmd([[
hi Normal guibg=NONE ctermbg=NONE
hi NormalNC guibg=NONE ctermbg=NONE
hi SignColumn guibg=NONE ctermbg=NONE
hi StatusLine guibg=NONE ctermbg=NONE
hi StatusLineNC guibg=NONE ctermbg=NONE
hi VertSplit guibg=NONE ctermbg=NONE
hi TabLine guibg=NONE ctermbg=NONE
hi TabLineFill guibg=NONE ctermbg=NONE
hi TabLineSel guibg=NONE ctermbg=NONE
hi Pmenu guibg=NONE ctermbg=NONE
hi PmenuSel guibg=NONE ctermbg=NONE
hi NeoTreeNormal guibg=NONE ctermbg=NONE
hi NeoTreeNormalNC guibg=NONE ctermbg=NONE
hi NeoTreeWinSeparator guibg=NONE ctermbg=NONE
hi NeoTreeEndOfBuffer guibg=NONE ctermbg=NONE
hi EndOfBuffer guibg=NONE ctermbg=NONE
]])
end

-- Apply transparency settings initially
set_transparency()

-- Reapply transparency on buffer enter
vim.api.nvim_create_autocmd("BufEnter", {
  pattern = "*",
  callback = set_transparency,
})

```

**Reply by lu5her-s** [reply]
why my which-key and lazy-git not transparent

**Comment by zodman** [comment]
I can suggest use this plugin:

https://github.com/xiyaowong/transparent.nvim
